corehq/apps/analytics/static/analytics/js/logging.js

```diff
@@ -0,0 +1,176 @@
+/* globals _, JSON */
```

Comment:
Would it be worth having some overall docs? There's a lot of functionality in here, might be helpful to have a couple of examples to demonstrate using different logging levels and groups (even after reading the full PR, I don't quite follow whether groups are something that the calling code uses, or something used internally to format data/arguments in messages better).

Reply:
groups are something internal to format data/argument messages better